pallets/staking/src/lib.rs

```diff
@@ -279,6 +299,37 @@ pub mod pallet {
             Ok(().into())
         }
 
+        /// Record the staking slash event, no real transfer happened.
+        /// TODO restrict the times an account can report in one day and max slash.
+        ///
+        /// May only be called from `T::WithdrawOrigin`.
+        ///
+        /// - `agent`: the multisig account of relay chain.
+        /// - `amount`: the rewarded assets.
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn record_slash(
+            origin: OriginFor<T>,
+            agent: T::AccountId,
```

Comment:
Can we remove this? We are not using this except sending in event.

Reply:
If no agent, there will no info to track which agent introduce slash, now we use the event to keep this info, but can be stored in storage in the future if need.